test/julia/Matrix-test.jl

```diff
@@ -0,0 +1,39 @@
+@testset "Julia.Matrix.manipulation" begin
+  r, c = 1, 2
+  A = zero_matrix(Int, r, c)
+  A[1, :] = [10, -2]
+  B = Int[10 -2]
+
+  @test A == B
+  @test number_of_rows(A) == r
+  @test number_of_columns(A) == c
+end
+
+@testset "Julia.Matrix.array_creation" begin
+  M = matrix_ring(ZZ, 2)
+  m1 = M(ZZ[1 2; 3 4])
+  m2 = M(ZZ[1 0; 0 1])
+  arr = Array(M, 2)
+  arr[1] = deepcopy(m1)
+  arr[2] = deepcopy(m2)
+  @test arr[1] == m1
+  @test arr[2] == m2
+end
+
+@testset "Julia.Matrix.permutations" begin
+  M = matrix_ring(ZZ, 2)
+  m0 = M([1 2; 3 4])
+  m1 = M([2 3; 4 5])
+  m2 = M([3 4; 5 6])
+  m3 = M([4 5; 6 7])
+  m = Matrix{elem_type(M)}(undef, 2, 2)
+  Rm = Matrix{elem_type(M)}(undef, 2, 2)
+  Cm = Matrix{elem_type(M)}(undef, 2, 2)
+  m[1, 1], m[1, 2], m[2, 1], m[2, 2] = m0, m1, m2, m3
+  Rm[1, 1], Rm[1, 2], Rm[2, 1], Rm[2, 2] = m2, m3, m0, m1
+  Cm[1, 1], Cm[1, 2], Cm[2, 1], Cm[2, 2] = m1, m0, m3, m2
```

Comment:
I wanted to suggest this:
```suggestion
  m = matrix(M, [m0  m1 ; m2  m3])
  Rm = matrix(M, [m2  m3 ; m0  m1])
  Cm = matrix(M, [m1  m0 ; m3  m2])
```
but it doesn't work as I expect it to work, which might be a bug in itself... but for now this should work:
```suggestion
  m = matrix(M, 2, 2,[m0, m1, m2, m3])
  Rm = matrix(M, 2, 2,[m2, m3, m0, m1])
  Cm = matrix(M, 2, 2,[m1, m0, m3, m2])
```

Reply:
Note that this will produce different types: in your code:
```
julia> typeof(m)
Matrix{MatRingElem{BigInt}} (alias for Array{AbstractAlgebra.Generic.MatRingElem{BigInt}, 2})
```
versus mine:
```
julia> typeof(m)
AbstractAlgebra.Generic.MatSpaceElem{AbstractAlgebra.Generic.MatRingElem{BigInt}}
```
Making this change will fix the test failure. But the test failure suggests there is a bug. Indeed we need this fix:
```patch
diff --git a/src/julia/Matrix.jl b/src/julia/Matrix.jl
index f636af1d1..a28183bb5 100644
--- a/src/julia/Matrix.jl
+++ b/src/julia/Matrix.jl
@@ -112,7 +112,7 @@ end
 function swap_cols!(a::Matrix{T}, i::Int, j::Int) where T <: NCRingElement
    if i != j
       for k = 1:nrows(a)
-         a[k, i], a[k, j] = a[k, i], a[k, j]
+         a[k, i], a[k, j] = a[k, j], a[k, i]
       end
    end
    return a
```
and then I guess we want a test case for that as well...

(Feel free to extend this PR to cover that, or else we'll fix it separately)

Reply:
The syntax `[A B; C D]` with Nemo matrices is ambiguous, right? We decided that it means forming the block matrix and not yielding a julia Matrix.

Reply:
Ops, why did you suggest a change 😠  Now I committed it because I thought it was okay!!1!1! 😡